backend/src/queries/channels.rs

```diff
@@ -0,0 +1,96 @@
+use sqlx::PgPool;
+use crate::errors::CustomError;
+use crate::models::{
+    database::Db,
+    channel::ChannelModel
+};
+
+impl Db {
+    pub async fn list_channels(pool: &PgPool) -> Result<Vec<ChannelModel>, CustomError> {
+        let channels = sqlx::query_as!(
+            ChannelModel,
+            "SELECT * FROM channel"
+        )
+        .fetch_all(pool)
+        .await
+        .map_err(|e| {
+            eprintln!("Database error: {}", e);
+            CustomError::DatabaseError(e.to_string())
+        })?;
```

Comment:
I'm also not sure we should do ANY error handling in these methods.  I think these should be the sql and only the sql in here and leave all other handling to.... wait for it.... the handlers.

Reply:
We want to do this because what we are gaining by doing this refactor is the ability to more easily unit test our handlers. We want to test the error handling but if we mock away these functions like `db::list_channels` then we cannot test the error handling.  Which is an important thing to test for this IMO because it directly determines response status codes.

Reply:
**s** _a_  _S_ **_s_**